packages/compass-import-export/src/components/export-select-fields.tsx

```diff
@@ -0,0 +1,261 @@
+
+    setAutoScrollNewFieldInput(false);
+  }, [autoScrollNewFieldInput]);
+
+  useEffect(() => {
+    if (
+      lastRenderedFieldsLength.current !== 0 &&
+      fieldKeys.length > lastRenderedFieldsLength.current
+    ) {
+      setAutoScrollNewFieldInput(true);
+    }
+
+    lastRenderedFieldsLength.current = fieldKeys.length;
+  }, [fieldKeys]);
+
+  return (
+    <>
+      <div className={headerContainerStyles}>
+        <Body weight="medium">Select Fields</Body>
+        <Body>
+          The fields displayed are from a sample of documents in the collection.
+          To ensure all fields are exported, add missing field names.
+        </Body>
+        <div>
+          <Button
+            variant="primary"
+            leftGlyph={<Icon glyph="Plus" />}
+            size="xsmall"
+            onClick={onAddNewFieldButtonClicked}
+          >
+            Add new field
+          </Button>
+        </div>
+      </div>
+
+      <div className={tableContainerStyles}>
+        <Table
+          data={fieldKeys.map((field, index) => ({
+            checked: !!fields[field],
+            field,
+            index,
+          }))}
+          columns={[
+            <TableHeader
+              className={smallCellContainerStyle}
+              key="checkbox"
+              label={
+                <Checkbox
+                  aria-label={
+                    isEveryFieldChecked
+                      ? 'Deselect all fields'
+                      : 'Select all fields'
+                  }
+                  title={
+                    isEveryFieldChecked
+                      ? 'Deselect all fields'
+                      : 'Select all fields'
+                  }
+                  checked={isEveryFieldChecked}
+                  onChange={handleHeaderCheckboxChange}
+                />
+              }
+            />,
+            <TableHeader key="field-name" label="Field Name" />,
+          ]}
+        >
+          {({ datum: field }) => (
+            <>
+              <Row key={field.field}>
+                <Cell className={smallCellContainerStyle}>
+                  <div className={checkboxContainerStyle}>
+                    <Checkbox
+                      title={`${field.checked ? 'Exclude' : 'Include'} ${
+                        field.field
+                      } in exported collection`}
+                      aria-label={`${field.checked ? 'Exclude' : 'Include'} ${
+                        field.field
+                      } in exported collection`}
+                      checked={field.checked}
+                      name={field.field}
+                      onChange={handleFieldCheckboxChange}
+                    />
+                  </div>
+                </Cell>
+                <Cell className={cellContainerStyles}>
+                  <Body>{field.field}</Body>
+                </Cell>
+              </Row>
+              {field.index === fieldKeys.length - 1 && (
+                <Row key=".__add-new-field">
```

Comment:
Took me a while to get that this is actually returning two `Row`s for one `datum`! Clever! :)